packages/gatsby/src/schema/build-node-types.js

```diff
@@ -160,6 +175,17 @@ module.exports = async ({ parentSpan }) => {
       fields: mergedFieldsFromPlugins,
     })
 
+    const inferredFieldNames = _.keys(
+      inferredInputFieldsFromPlugins.inferredFields
+    )
+
+    // Track which fields are supplied by plugins. This will determine
+    // if we can query by pure data or whether we have to resolve the
+    // fields first. See `./run-query.js`
+    _.forEach(inferredFieldNames, fieldName => {
+      pluginFieldTracking.add(fieldName)
```

Comment:
I think we should namespace fieldNames - if one node type uses resolver for field `foo`, we don't want to revert to `sift` if we query other type and use field with same `foo` name